### ISSUE TITLE
custom formatters for dynamic columns

### ISSUE THREAD

**Issue by ray007** [arnedesmedt/vue-ads-table-tree#18]
I have a dynamic column configuration loaded from the server. Since the column config does not support a custom formatter, I tried to get one working with the templates. My current attempt, which still does not work, looks like this:
```xml
<vue-ads-table ref="adsTable" :columns="tableCols" :rows="tableData">
	<template slot="no-rows">No data</template>
	<template v-if="tableCols.length">
		<template v-for="column in tableCols">
			<template v-if="column.formatter" slot="{{column.property}}" slot-scope="props">{{
				column.formatter(props.row)
			}}</template>
		</template>
	</template>
</vue-ads-table>
```
Any hints what to do to get this working?

**Comment by arnedesmedt** [comment]
Hi,

What do you mean by a custom formatter? You can use the column slot identified by the column property. See https://github.com/arnedesmedt/vue-ads-table-tree#slots 

And are you using the full component or the table component?

**Comment by ray007** [comment]
By "full component" you mean the tree table? Yes, that's the one I'm trying to use.

I'm loading a view config object from the server at runtime, and from this I'm generating a column configuration at runtime.
Since I do not know the available columns in advance, I cannot write just a template with a fixed value in the `slot` attribute. Which is why I tried to define a formatting function on the column definition object when needed and call that.
Maybe I just know enough about vuejs yet, but so far I haven't managed to find a syntax to get my formatter called.

**Comment by arnedesmedt** [comment]
Can you put here or on jsfiddle/codepen an example of your code? 
You can btw also write a for loop to generate the cell/column slots at runtime while receiving the column configuration?

**Comment by ray007** [comment]
Write a loop to generate the columns slots at runtime is exactly what I'm trying (and failing) to do.
My current attempt looks like this:
```xml
<template v-if="tableCols.length">
    <template v-for="column in tableCols">
        <template v-if="column.formatter" :slot="column.property" slot-scope="props">{{
            column.formatter(props.row)
        }}</template>
    </template>
</template>
```
But my `formatter` function never gets called.
Or do you sanitize the column config objects and I need to put the formatters somewhere else keyed by id?

**Comment by arnedesmedt** [comment]
Try to combine all your templates in one tag. I've worked out an example:

https://jsfiddle.net/arnedesmedt/knLv3tgs/

**Comment by ray007** [comment]
Hmm, after pulling `:slot` and `slot-scope` into the `v-for` template, our code now looks quite similar, sans pagination.
But if I edit your sample to set the row-data asyncronously in a timeout, I suddenly see "no data".
Will need to do more testing tomorrow...

**Comment by arnedesmedt** [comment]
I'll added a fix for this in 2.3.3
